File: paddler_types/src/agent_desired_model.rs
use serde::Deserialize;
use serde::Serialize;

use crate::huggingface_model_reference::HuggingFaceModelReference;
use crate::url_model_reference::UrlModelReference;

#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(deny_unknown_fields)]
pub enum AgentDesiredModel {
    HuggingFace(HuggingFaceModelReference),
    LocalToAgent(String),
    Url(UrlModelReference),
    #[default]
    None,
}

File: paddler_client_javascript/src/schemas/AgentDesiredModel.ts
import { z } from "zod";

import { HuggingFaceModelReferenceSchema } from "./HuggingFaceModelReference";
import { UrlModelReferenceSchema } from "./UrlModelReference";

export const AgentDesiredModelSchema = z.union([
  z.object({
    HuggingFace: HuggingFaceModelReferenceSchema,
  }),
  z.object({
    LocalToAgent: z.string(),
  }),
  z.object({
    Url: UrlModelReferenceSchema,
  }),
  z.literal("None"),
]);

export type AgentDesiredModel = z.infer<typeof AgentDesiredModelSchema>;
